src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+                return;
+        }
+    } else {
+        switch (arg0) {
+            case 2:
+                func_800BA808(gMsg_ID_20015, 0x14);
+                return;
+            case 3:
+                func_800BA808(gMsg_ID_20016, 0x1E);
+                return;
+        }
+    }
+}
+
+void func_80048CC4(Object_2F4* arg0, s32 arg1) {
+    Object_2F4_Initialize(arg0);
+    arg0->obj.status = 1;
+    arg0->obj.id = 0xC3;
+    arg0->obj.pos.x = D_800C9F60[arg1].x + gPlayer[0].pos.x;
+    arg0->obj.pos.y = D_800C9F60[arg1].y + gPlayer[0].pos.y;
+    arg0->obj.pos.z = D_800C9F60[arg1].z + gPlayer[0].unk_138;
+    arg0->unk_0F4.y = 0.0f;
+    arg0->unk_0F4.z = D_800C9F90[arg1];
+    Object_SetInfo(&arg0->info, arg0->obj.id);
+    if (arg1 == 3) {
+        Audio_PlaySfx(0x11030010U, &arg0->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        Audio_PlaySfx(0x31024059U, &arg0->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        arg0->unk_0B6 = 1;
+    } else {
+        arg0->unk_07C = 1;
+        Audio_PlaySfx(0x3100000CU, &arg0->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+    }
+}
+
+void func_80048E40(Player* arg0) {
+    Vec3f sp5C;
+    Vec3f sp50;
+    s8 pad[0x8];
+    f32 temp_ft3;
+    f32 temp_ret;
+    s32 temp_v0;
+    s32 temp_v1;
+
+    gObjects408[1].obj.status = 0;
+    Math_SmoothStepToF(&arg0->unk_130, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_12C, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0E4, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0E8, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_0EC, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_08C, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_034, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToAngle(&arg0->unk_4D8, 0.0f, 0.1f, 20.0f, 0.0f);
+    Math_SmoothStepToF(&arg0->unk_110, 0.0f, 0.1f, 3.0f, 0.0f);
+
+    switch (arg0->unk_1D0) {
+        case 0:
+            arg0->unk_4DC = 0;
+            func_8001A38C(1, &arg0->unk_460);
+            arg0->unk_1D0 += 1;
+            D_80177A48[0] = 0.0f;
+            D_80177A48[1] = 0.0f;
+            D_80177A48[2] = 0.0f;
+            arg0->wings.unk_2C = 1;
+            D_80177A80 = 0;
+
+        case 1:
+            D_80177A48[1] -= D_80177A48[2];
+            Math_SmoothStepToF(&D_80177A48[2], 1.2f, 0.1f, 0.01f, 0.0f);
+            Math_SmoothStepToF(D_80177A48, 0.1f, 0.1f, 0.002f, 0.0f);
+            Math_SmoothStepToF(&arg0->unk_0D0, 0.0f, 1.0f, 0.5f, 0.0f);
+            Matrix_RotateX(gCalcMatrix, -0.08726646f, 0);
+            Matrix_RotateY(gCalcMatrix, D_80177A48[1] * M_DTOR, 1);
+            sp5C.x = 0.0f;
+            sp5C.y = 0.0f;
+            sp5C.z = 300.0f;
+            Matrix_MultVec3f(gCalcMatrix, &sp5C, &sp50);
+            D_801779A0 = arg0->pos.x;
+            D_801779B8 = arg0->pos.y;
+            D_801779C0 = arg0->unk_138 + D_80177D20;
+
+            switch (D_80177A80) {
```

Comment:
From what I can tell, `D_80177A80` is a frame counter used for timing cutscene events. Comparisons involving it should use decimal.